cascade-api/test/Test/Cascade/Api/StateMachine/Command/Task.hs

```diff
@@ -0,0 +1,481 @@
+getAllByProjectIdForNonExistingProject =
+  let
+    generator :: Model Symbolic -> Maybe (g (GetAllByProjectId Symbolic))
+    generator model = case model ^. #project . #notExistingIds of
+      []         -> Nothing
+      projectIds -> Just $ do
+        projectId <- Gen.element projectIds
+        pure $ GetAllByProjectId { .. }
+
+    execute :: GetAllByProjectId Concrete -> m Cascade.Api.Projects.Tasks.GetAllByProjectIdResponse
+    execute (GetAllByProjectId projectId) = evalIO $ Cascade.Api.Projects.Tasks.getAllByProjectId (concrete projectId)
+
+    ensure :: Model Concrete -> Model Concrete -> GetAllByProjectId Concrete -> Cascade.Api.Projects.Tasks.GetAllByProjectIdResponse -> Test ()
+    ensure _before _after _input response = do
+      footnoteShow response
+
+      response ^. #responseStatusCode . #statusCode === 404
+  in
+    Command generator execute [Ensure ensure]
+
+
+-- brittany-disable-next-binding
+data AddNotExistingId (v :: Type -> Type) = AddNotExistingId
+  { id :: Task.Id
+  }
+  deriving stock Show
+
+instance HTraversable AddNotExistingId where
+  htraverse _ (AddNotExistingId id) = pure $ AddNotExistingId id
+
+addNotExistingId :: forall g m . MonadGen g => Applicative m => Command g m Model
+addNotExistingId =
+  let generator :: Model Symbolic -> Maybe (g (AddNotExistingId Symbolic))
+      generator _ = Gen.id |> fmap AddNotExistingId |> Just
+
+      execute :: AddNotExistingId Concrete -> m Task.Id
+      execute (AddNotExistingId id) = pure id
+
+      update :: Model v -> AddNotExistingId v -> Var Task.Id v -> Model v
+      update model _input id = model |> #task . #notExistingIds %~ cons id
+  in  Command generator execute [Update update]
+
+-- brittany-disable-next-binding
+newtype GetById (v :: Type -> Type) = GetById
+  { id :: Var Task.Id v
+  }
+  deriving stock (Generic, Show)
+
+instance HTraversable GetById where
+  htraverse f (GetById id) = GetById <$> htraverse f id
+
+getExistingById :: forall g m . MonadGen g => MonadIO m => MonadTest m => Command g m Model
+getExistingById =
+  let generator :: Model Symbolic -> Maybe (g (GetById Symbolic))
+      generator model = case model ^.. #task . #creatables . folded . to Map.keys |> mconcat of
+        []  -> Nothing
+        ids -> Gen.element ids |> fmap GetById |> Just
+
+      execute :: GetById Concrete -> m Cascade.Api.Tasks.GetByIdResponse
+      execute (GetById id) = evalIO . Cascade.Api.Tasks.getById $ concrete id
+
+      ensure :: Model Concrete -> Model Concrete -> GetById Concrete -> Cascade.Api.Tasks.GetByIdResponse -> Test ()
+      ensure before _after (GetById taskId) response = do
+        footnoteShow response
+
+        task :: Task.Readable <- (response ^. #responseBody) |> matchUnion @(Response.Ok Task.Readable) |> coerce |> evalMaybe
+
+        let id = task ^. #id
+
+        concrete taskId === id
```

Comment:
This assignment looks redundant. We are actually using the `Var` form of id in the next line query:

``` haskell
(Var $ Concrete id)
```

Which is exactly the `taskId`.